src/theory/sets/theory_sets_private.cpp

```diff
@@ -730,10 +730,10 @@ void TheorySetsPrivate::checkFilterUp()
       Node p_x = nm->mkNode(Kind::APPLY_UF, p, x);
       Node skolem = d_treg.getProxy(term);
       Node memberFilter = nm->mkNode(Kind::SET_MEMBER, x, skolem);
-      Node not_p_x = p_x.notNode();
-      Node not_memberFilter = memberFilter.notNode();
-      Node orNode =
-          p_x.andNode(memberFilter).orNode(not_p_x.andNode(not_memberFilter));
+      // (set.member x A)
+      // ---------------------------------------
+      // (set.member x (set.filter P A)) = (P x)
+      Node orNode = memberFilter.eqNode(p_x);
```

Comment:
```suggestion
      Node conclusion= memberFilter.eqNode(p_x);
```
Can you update the documentation in header file as well?